mindmeld/augmentation.py

```diff
@@ -0,0 +1,315 @@
+            model_name, force_download=False
+        ).to(torch_device)
+
+        # Use default model params if not customized for app
+        self.params = params or {
+            "max_length": 60,
+            "num_beams": 10,
+            "num_return_sequences": 10,
+            "temperature": 1.5,
+        }
+
+    def _get_response(self, query):
+        """Generates paraphrase responses for given query.
+
+        Args:
+            query (str): An application query.
+
+        Return:
+            paraphrases (list(str)): List of paraphrased queries.
+        """
+        batch = self.tokenizer.prepare_seq2seq_batch(
+            [query], truncation=True, padding="longest", max_length=60
+        )
+        translated = self.model.generate(
+            **batch,
+            **self.params,
+        )
+        paraphrases = self.tokenizer.batch_decode(translated, skip_special_tokens=True)
+        return paraphrases
+
+    def augment_queries(self, queries, **kwargs):
+        augmented_queries = []
+        for query in queries:
+            augmented_queries.extend(self._get_response(query, **kwargs))
+        return augmented_queries
+
+
+class MultiLingualParaphraser(Augmentor):
+    """Paraphraser class for generating paraphrases based on language code of the app
+    (currently supports: French, Italian, Spanish, Portuguese, and Romanian).
+    """
+
+    def __init__(self, lang, paths, path_suffix, params, resource_loader):
+        """Initializes a multi-lingual paraphraser.
+
+        Args:
+            lang (str): The lang code for paraphrasing
+            paths (list): Path rules for fetching relevant files to Paraphrase.
+            path_suffix (str): Suffix to be added to new augmented files.
+            params (dict): Model params.
+            resource_loader (object): Resource Loader object for the application.
+        """
+        super().__init__(
+            lang=lang,
+            paths=paths,
+            path_suffix=path_suffix,
+            resource_loader=resource_loader,
+        )
+
+        self.torch_device = (
+            "cuda" if _get_module_or_attr("torch.cuda", "is_available")() else "cpu"
+        )
+
+        MarianTokenizer = _get_module_or_attr("transformers", "MarianTokenizer")
+        MarianMTModel = _get_module_or_attr("transformers", "MarianMTModel")
+
+        en_model_name = "Helsinki-NLP/opus-mt-ROMANCE-en"
+        self.en_tokenizer = MarianTokenizer.from_pretrained(en_model_name)
+        self.en_model = MarianMTModel.from_pretrained(en_model_name)
+        self.en_model.to(self.torch_device)
+
+        target_model_name = "Helsinki-NLP/opus-mt-en-ROMANCE"
+        self.target_tokenizer = MarianTokenizer.from_pretrained(target_model_name)
+        self.target_model = MarianMTModel.from_pretrained(target_model_name).to(
+            self.torch_device
+        )
+
+        # Use default model params if not customized for app
+        self.params = params or {
```

Comment:
Though we agreed on having two sets of parameters here. One for the pivot language translation and the other for the "back-translation".